main.py

```diff
@@ -11,9 +13,39 @@
 from models import get_engine, get_session
 
 
+def base_filter(giving_partner, active, unregistered):
+    "base filter to reuse in SELECT queries to retrieve GPs from donee_info DB"
+    return [
+        giving_partner.active == active,
+        giving_partner.unregistered == unregistered,
+        giving_partner.country.isnot(None),
+        func.trim(giving_partner.country) != "",
+    ]
+
+
+def parse_args():
+    """function to parse optional giving partner id command line argument"""
+    parser = argparse.ArgumentParser(description="optional giving partner id")
+    parser.add_argument("--id", type=int)
+    try:
+        args = parser.parse_args()
+    except SystemExit:
+        print(
+            "parsing --id argument failed: please make sure it is an integer"
+        )  # error log this
+        raise
+    return args
+
+
 def main():
     """Main module"""
 
+    try:
+        args = parse_args()
+    except SystemExit:
+        print("parsing args failed")
+        raise
+
```

Comment:
You can just combine all the initialization logic into one try block.


```suggestion
    try:
        args = parse_args()

        engine = get_engine(
            db_host=Config.DB_HOST,
            db_port=Config.DB_PORT,
            db_user=Config.DB_USER,
            db_password=Config.DB_PASSWORD,
            db_name=Config.DB_NAME,
        )
        # log success
    except SystemExit:
        print("parsing args failed")
        raise
    except SQLAlchemyError as e:
        print(f"Failed to initialize database engine: {e}")  # error log this
        raise

```